components/director/internal/domain/application/service.go

```diff
@@ -488,6 +497,35 @@ func (s *service) DeleteLabel(ctx context.Context, applicationID string, key str
 	return nil
 }
 
+// IsBeingUsed Checks if an application has scenarios associated with it. If the scenarios are part of a runtime, then the application is considered being used by that runtime.
+func (s *service) IsBeingUsed(ctx context.Context, tenant, id string) (bool, error) {
+	scenarios, err := s.getScenarioNamesForApplication(ctx, id)
+	if err != nil {
+		return false, err
+	}
+
+	validScenarios := removeDefaultScenario(scenarios)
+	if len(validScenarios) > 0 {
+		runtimes, err := s.getRuntimeNamesForScenarios(ctx, tenant, validScenarios)
+		if err != nil {
+			return false, err
+		}
+
+		if len(runtimes) > 0 {
+			application, err := s.appRepo.GetByID(ctx, tenant, id)
+			if err != nil {
+				return true, errors.Wrapf(err, "while getting application with id %s", id)
+			}
+			msg := fmt.Sprintf("System %s is still used and cannot be deleted. Unassign the system from the following formations first: %s. Then, unassign the system from the following runtimes, too: %s", application.Name, strings.Join(validScenarios, ", "), strings.Join(runtimes, ", "))
```

Comment:
im pretty sure i've seen this code somewhere else, maybe in the delete app flow ? If its copied can we rather try to extract and reuse in order to avoid double maintanance in case this  code evolves in the future ?